flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierAligner.java

```diff
@@ -69,10 +79,14 @@
 	CheckpointBarrierAligner(
 			int totalNumberOfInputChannels,
```

Comment:
nit: this argument seems redundant because it can be got from `channelIndexToInputGate.length`